WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -55,4 +65,190 @@ class WordPressAuthenticatorTests: XCTestCase {
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testSignInForWPOrgReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPOrg()
+
+        XCTAssertTrue(vc is LoginSiteAddressViewController)
+    }
+
+    func testShowLoginForJustWPComSetsMetaProperties() throws {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy,
+                                                     xmlrpc: "https://example.com/xmlrpc.php",
+                                                     username: "username",
+                                                     connectedEmail: "email-address@example.com")
+
+        let navController = try XCTUnwrap(presenterSpy.presentedVC as? LoginNavigationController)
+        let controller = try XCTUnwrap(navController.viewControllers.first as? LoginEmailViewController)
+
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertEqual(controller.loginFields.restrictToWPCom, true)
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogXMLRPC, "https://example.com/xmlrpc.php")
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogUsername, "username")
+        XCTAssertEqual(controller.loginFields.username, "email-address@example.com")
+    }
+
+    func testShowLoginForSelfHostedSitePresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testSignInForWPComReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPCom()
+
+        XCTAssertTrue((vc as Any) is LoginEmailViewController)
+    }
+
+    func testSignInForWPComWithLoginFieldsReturnsVC() throws {
+        let navController = try XCTUnwrap(WordPressAuthenticator.signinForWPCom(dotcomEmailAddress: "example@email.com", dotcomUsername: "username") as? UINavigationController)
+        let vc = navController.topViewController
+
+        XCTAssertTrue(vc is LoginWPComViewController)
+    }
+
+    func testSignInForWPComSetsEmptyLoginFields() {
+        let navController = WordPressAuthenticator.signinForWPCom(dotcomEmailAddress: nil, dotcomUsername: nil) as! UINavigationController
+        let vc = navController.topViewController as! LoginWPComViewController
```

Comment:
How do you feel about using `try XCTUnwrap` and `as?` here, like you did in a few tests above?

If we can prevent the tests from crashing, we'll get a failure report with more information.